vochain/state.go

```diff
@@ -79,33 +81,50 @@ type ImmutableState struct {
 func NewState(dataDir string) (*State, error) {
 	var err error
 	vs := &State{}
-	//vs.Store = new(gravitonstate.GravitonState)
-	vs.Store = new(iavlstate.IavlState)
-
-	if err = vs.Store.Init(dataDir, "disk"); err != nil {
-		return nil, err
-	}
-
-	if err := vs.Store.AddTree(AppTree); err != nil {
-		return nil, err
-	}
-	if err := vs.Store.AddTree(ProcessTree); err != nil {
-		return nil, err
+	if err := initStore(dataDir, vs); err != nil {
+		return nil, fmt.Errorf("cannot init db: %s", err)
 	}
-	if err := vs.Store.AddTree(VoteTree); err != nil {
-		return nil, err
-	}
-
 	// Must be -1 in order to get the last committed block state, if not block replay will fail
 	if err = vs.Store.LoadVersion(-1); err != nil {
-		return nil, err
+		if err == iavl.ErrVersionDoesNotExist {
```

Comment:
What about moving this snippet to the `statedb/iavltree` implementation? It could be added to the `LoadVersion()` method. Then we would make this only executable if IAVL is actually used.

Reply:
Tried that, but I am getting weird errors like:
- Cannot init state, resource temporary unavailable.
- Oracle already added (On init chain), which is a fatal.
- Same as commented before but for validators.

None of this errors is raised going back to the original code, I can spend some time investigating this weird behavior or we could just merge this and we can tackle this afterwards.